packages/integration-sdk-runtime/src/storage/FileSystemGraphObjectStore/FileSystemGraphObjectStore.ts

```diff
@@ -56,6 +58,32 @@ export class FileSystemGraphObjectStore {
     }
   }
 
+  async getEntity(
+    getEntityOptions: GraphObjectLookupKey,
+  ): Promise<Entity> {
+    const { _type, _key } = getEntityOptions;
+    await this.flushEntitiesToDisk();
+
+    const entities: Entity[] = [];
+    await this.iterateEntities(
+      { _type, },
+      async (e) => {
+        if (e._key === _key) {
+          entities.push(e);
+        }
+      }
+    );
+
+    if (entities.length !== 1) {
+      throw new IntegrationError({
+        code: 'GET_ENTITY_ERROR',
+        message: `Failed to find entity with _type=${_type}, _key=${_key}`,
+      });
```

Comment:
I think it would be good to consider throwing the [existing duplicate key error type](https://github.com/JupiterOne/sdk/blob/0cb90d3c432902870b0ca2b4d11794ec9482ea4d/packages/integration-sdk-core/src/errors.ts#L146).

Reply:
Sounds good. I broke this up to throw `IntegrationMissingKeyError` when length === 0 and `IntegrationDuplicateKeyError` when length > 1.